test/integration/ratelimit_integration_test.cc

```diff
@@ -167,13 +169,23 @@ TEST_P(RatelimitIntegrationTest, Error) {
 TEST_P(RatelimitIntegrationTest, Timeout) {
   initiateClientConnection();
   waitForRatelimitRequest();
-  test_server_->waitForCounterGe("cluster.ratelimit.upstream_rq_timeout", 1);
+  switch (clientType()) {
+  case Grpc::ClientType::EnvoyGrpc:
+    test_server_->waitForCounterGe("cluster.ratelimit.upstream_rq_timeout", 1);
+    EXPECT_EQ(1, test_server_->counter("cluster.ratelimit.upstream_rq_timeout")->value());
+    EXPECT_EQ(1, test_server_->counter("cluster.ratelimit.upstream_rq_504")->value());
+    break;
+  case Grpc::ClientType::GoogleGrpc:
+    test_server_->waitForCounterGe("grpc.ratelimit.streams_closed_4", 1);
+    EXPECT_EQ(1, test_server_->counter("grpc.ratelimit.streams_total")->value());
+    EXPECT_EQ(1, test_server_->counter("grpc.ratelimit.streams_closed_4")->value());
```

Comment:
"grpc.ratelimit.streams_closed_4" is a pretty strange stat name. Is that a bug?

Reply:
Nup; it's `grpc.<stat prefix>.streams_closed_<grpc status code>`.

Reply:
Where is this implemented? I'm confused. Why don't we have similar stats for Envoy gRPC client?

Reply:
We had a discussion about this, where the outcome was that this would make more sense to do in a router related stat for Envoy gRPC. It will be implemented in the Google gRPC PR coming up, where with Google gRPC we maintain stats that would normally show up under upstream cluster stats etc.

Reply:
Ah OK, my main question was how this is passing, since I didn't see this implemented yet. So this PR doesn't actually enable the Google client tests yet?

Reply:
Nevermind, I see the parameter only running Envoy tests now. OK.